include/ginkgo/core/base/abstract_factory.hpp

```diff
@@ -435,14 +443,15 @@ class deferred_factory_parameter {
  * @ingroup LinOp
  */
 #define GKO_FACTORY_PARAMETER(_name, ...)                                    \
-    mutable _name{__VA_ARGS__};                                              \
+    _name{__VA_ARGS__};                                                      \
```

Comment:
I guess now this cannot be modified from a `const Factory` ?

Reply:
So, `with_*` returns `self()` , which can be `const` or non-const. `.on()` also has been changed to return `FactoryType` which is always non-const ?

Reply:
No, because `with_*` is not const function, it can only be called by non-const and return non-const. 
Yes for last question, `.on()` always return `FactoryType`, which is a different type than `FactoryType::parameters_type`